pkg/kube/kclient/client.go

```diff
@@ -333,17 +333,16 @@ func applyDynamicFilter[T controllers.ComparableObject](filter Filter, gvr schem
 				// Namespace is special; we query all namespaces
 				// Note: other cluster-scoped resources should just not use the filter
 				for _, item := range ic.ListUnfiltered(metav1.NamespaceAll, klabels.Everything()) {
-					if !added.Contains(item.GetName()) && !removed.Contains(item.GetName()) {
+					if !added.Contains(item.GetName()) {
 						continue
 					}
 					for _, c := range ic.registeredHandlers {
-						if added.Contains(item.GetName()) {
-							c.handler.OnAdd(item, false)
-						} else {
-							c.handler.OnDelete(item)
-						}
+						c.handler.OnAdd(item, false)
 					}
 				}
+				// Removes are currently NOT handled. We only have the namespace name here. We would need to have the object
+				// filter passthrough the entire namespace object, so we can pass the last known state to OnDelete.
+				// Fortunately, missing a namespace delete event usually doesn't matter since everything in the namespace gets torn down.
```

Comment:
Does this mean there is no namespace delete triggered? I think this is not right, this package can be used by not only istiod, but also other thirdparty users

Reply:
That is the state even before this PR. Istio packages are not intended for usage outside istio so I don't really care much about the 3p use case? 

We can make it send deletes consistently but it's extremely hard to do so. The problem is the filter and the namespace handler ordering is random (from informer) so we cannot guarantee the order. I don't think the complexity is worth it imo. 

Either way IMO we could merge this even if we want to do this eventually, as it is strictly fixing the bug -- the missing deletes was already present